code-challenges.test.js

```diff
@@ -10,41 +19,181 @@
+// a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+const colors1 = ["purple", "blue", "green", "yellow", "pink"];
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
-
-
+const colors2 = [
+  "chartreuse",
+  "indigo",
+  "periwinkle",
+  "ochre",
+  "aquamarine",
+  "saffron",
+];
+//Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+
+describe("removeShuffle", () => {
+  const colors1 = ["purple", "blue", "green", "yellow", "pink"];
+  const colors2 = [
+    "chartreuse",
+    "indigo",
+    "periwinkle",
+    "ochre",
+    "aquamarine",
+    "saffron",
+  ];
+  it("takes in an array, removes the first item from the array and shuffles the remaining content.", () => {
+    expect(removeShuffle(colors1)).toEqual(
+      expect.arrayContaining(["yellow", "blue", "pink", "green"])
+    );
+    expect(removeShuffle(colors2)).toEqual(
+      expect.arrayContaining([
+        "saffron",
+        "aquamarine",
+        "periwinkle",
+        "indigo",
+        "ochre",
+      ])
+    );
+  });
+});
 // b) Create the function that makes the test pass.
-
+// First declare a new function named remove Shuffle which takes an array as a parameter
+// use the mutator method .shift to remove the first item in thr array
+// set up a new array empty array to move random elements into
+// set up a while loop that will iterate on the original array as long as it has a length of >0
+// use the splice method in addition to math.random to cut out items at a randomly generated index of the first array and move them into the second array.
+// splice returns the cutout items as an array. in order to add the items into the new array as strings, I will take the 0 index of the array which will just be the cut out element.
+//
+
+const removeShuffle = (array) => {
+  array.shift();
+  let newArray = [];
+  while (array.length > 0) {
+    newArray = [
+      ...newArray,
+      array.splice(Math.floor(Math.random() * array.length), 1)[0],
+    ];
+  }
+  return newArray;
+};
+
+//    Test Suites: 1 passed, 1 total
+//    Tests:       1 passed, 1 total
+
+//Note: when I comment out the code which removes the first element (shift), it still passes the test. Based on my reading of the arrayContaining documentation, it does not care if there are extra items in the array it receives. I found another way to test if the first item has been cut off and have added it below. I tried to incorporate it within the original test and wasn't able to find the syntax to do so.
```

Comment:
Nice work in thinking about how to make the test even more efficient.  Really cool approach to this problem!